src/components/User/Bookings.jsx

```diff
@@ -21,7 +21,19 @@ export default function Bookings() {
             console.error("Error while fetching bookings.", error);
         }
     }, [setBookings]);
-    
+
+    const confirmBooking = useCallback(async (bookingId) => {
+        try {
+            const response = await axios.get(`/api/user/bookings/confirm/${bookingId}`);
+            if (response.status === 200) {
+                // Remove deleted booking from state
+                setBookings(response.data);
+            }
+        } catch (error) {
+            console.error("Error confirming booking.", error);
```

Comment:
I think this all totally fine, but you could add a fallback like an "alert" window to indicate the user that something went wrong, to keep the user informed.